lib/nerves_network/dhclient.ex

```diff
@@ -68,31 +68,31 @@ defmodule Nerves.Network.Dhclient do
     ["-S"]
   end
 
-  defp append_ifname(input_str, ifname) do
-    input_str <> "." <> ifname
+  defp append_ifname(input_str, ifname, mode) do
+    input_str <> "." <> ifname <> "." <> to_string(mode)
   end
 
-  defp runtime_lease_file(ifname, runtime) do
+  defp runtime_lease_file(ifname, runtime, mode) do
       if Keyword.has_key?(runtime, :lease_file) do
-        [ "-lf", Keyword.get(runtime, :lease_file) |> append_ifname(ifname) ]
+        [ "-lf", Keyword.get(runtime, :lease_file) |> append_ifname(ifname, mode) ]
```

Comment:
well the lease file and pid file were supposed to be handled independently not sure whether appending ifname and mode to each wold not break things